Client/mods/deathmatch/logic/luadefs/CLuaPedDefs.cpp

```diff
@@ -2336,3 +2340,51 @@ int CLuaPedDefs::DetonateSatchels(lua_State* luaVM)
     lua_pushboolean(luaVM, false);
     return 1;
 }
+
+int CLuaPedDefs::SetPedEnterVehicle(lua_State* luaVM)
+{
```

Comment:
Please consider using the new argument parser.

Reply:
```cpp
bool CLuaPedDefs::SetPedEnterVehicle(CClientPed* pPed, CClientVehicle* pVehicle, std::optional<bool> bPassenger)
{
// use bPassenger like: bPassenger.value_or(false)
// Also, I see you used a default value for pVehicle, I presume thats a typo
// if not, use std::optional for it here
}
// Also, when adding the function to the header this prototype:
// static bool SetPedEnterVehicle(CClientPed* pPed, CClientVehicle* pVehicle, std::optional<bool> bPassenger)
```

Reply:
Uh, also, use `ArgumentParser` in the function list.
See `CLuaCryptDefs` for examples.

Reply:
Thx for the pointers, I've added it.